File: src/main/java/world/estaria/paper/dependency/loader/DependencyPluginLoader.java
package world.estaria.paper.dependency.loader;

import io.papermc.paper.plugin.loader.PluginClasspathBuilder;
import io.papermc.paper.plugin.loader.PluginLoader;
import io.papermc.paper.plugin.loader.library.impl.JarLibrary;
import org.jetbrains.annotations.NotNull;

import java.io.File;
import java.util.Arrays;
import java.util.List;

public class DependencyPluginLoader implements PluginLoader {

    @Override
    public void classloader(@NotNull PluginClasspathBuilder classpathBuilder) {
        File directory = new File("dependencies/");
        List<File> list = Arrays.stream(directory.listFiles())
                .filter(file -> file.getName().endsWith(".jar"))
                .toList();

        for (File file : list) {
            System.out.println("Loading dependency " + file.getName());
            classpathBuilder.addLibrary(new JarLibrary(file.toPath()));
        }
    }

}

File: src/main/java/world/estaria/paper/dependency/loader/DependencyPluginLoader.kt
package world.estaria.paper.dependency.loader

import io.papermc.paper.plugin.loader.PluginClasspathBuilder
import io.papermc.paper.plugin.loader.PluginLoader
import io.papermc.paper.plugin.loader.library.impl.JarLibrary
import java.io.File

/**
 * @author Niklas Nieberler
 */

class DependencyPluginLoader : PluginLoader {

    override fun classloader(classpathBuilder: PluginClasspathBuilder) {
        val directory = File("dependencies/")
        val files = directory.listFiles()!!
            .filter { it.name.endsWith(".jar") }

        files.forEach {
            println("Loading dependency ${it.nameWithoutExtension}")
            classpathBuilder.addLibrary(JarLibrary(it.toPath()))
        }
    }

}